lib/middleware.js

```diff
@@ -131,6 +127,27 @@ module.exports = function wrapper(context) {
 
         resolve();
       }
+
+      if (
+        context.options.lazy &&
+        (!context.options.filename || context.options.filename.test(filename))
+      ) {
+        context.rebuild();
+      }
+
+      if (HASH_REGEXP.test(filename)) {
+        try {
+          if (context.fs.statSync(filename).isFile()) {
+            processRequest();
+
+            return;
+          }
+        } catch (_error) {
+          // Ignore error
+        }
+      }
+
+      ready(context, processRequest, req);
```

Comment:
Move this inside file, no need util for simple action, also it is hard to read, because you should look in two files when request happens and it is painful